File: VB/MainDemo.Module/BusinessObjects/Employee.vb
Imports System
Imports DevExpress.Xpo
Imports System.ComponentModel
Imports DevExpress.Persistent.Base
Imports System.Collections.Generic
Imports DevExpress.ExpressApp.Security.Strategy
Imports DevExpress.Persistent.Validation

Namespace MainDemo.Module.BusinessObjects
	<DefaultClassOptions, DefaultProperty("FullName"), ImageName("BO_User")>
	Public Class Employee
		Inherits SecuritySystemUser

		Private _LastName As String
		Private _FirstName As String
        Private _department As Department
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Public Property FirstName() As String
			Get
				Return _FirstName
			End Get
			Set(ByVal value As String)
				SetPropertyValue("FirstName", _FirstName, value)
			End Set
		End Property
		Public Property LastName() As String
			Get
				Return _LastName
			End Get
			Set(ByVal value As String)
				SetPropertyValue("LastName", _LastName, value)
			End Set
		End Property
		<PersistentAlias("concat(FirstName, ' ', LastName)")>
		Public ReadOnly Property FullName() As String
			Get
				Return Convert.ToString(EvaluateAlias("FullName"))
			End Get
		End Property
		<Association, RuleRequiredField>
		Public Property Department() As Department
			Get
                Return _department
			End Get
			Set(ByVal value As Department)
                SetPropertyValue("Department", _department, value)
			End Set
		End Property
		<Association>
		Public ReadOnly Property Tasks() As XPCollection(Of EmployeeTask)
			Get
				Return GetCollection(Of EmployeeTask)("Tasks")
			End Get
		End Property
	End Class
End Namespace

File: CS/MainDemo.Module/BusinessObjects/Employee.cs
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl.PermissionPolicy;
using DevExpress.Persistent.Validation;
using DevExpress.Xpo;
using System;
using System.ComponentModel;

namespace MainDemo.Module.BusinessObjects {
    [DefaultClassOptions]
    [DefaultProperty("FullName")]
    [ImageName("BO_User")]
    public class Employee : PermissionPolicyUser {
        private string _LastName;
        private string _FirstName;
        private Department department;
        public Employee(Session session) :
            base(session) {
        }
        public string FirstName {
            get {
                return _FirstName;
            }
            set {
                SetPropertyValue("FirstName", ref _FirstName, value);
            }
        }
        public string LastName {
            get {
                return _LastName;
            }
            set {
                SetPropertyValue("LastName", ref _LastName, value);
            }
        }
        [PersistentAlias("concat(FirstName, ' ', LastName)")]
        public string FullName {
            get {
                return Convert.ToString(EvaluateAlias("FullName"));
            }
        }
        [Association]
		[RuleRequiredField]
        public Department Department {
            get {
                return department;
            }
            set {
                SetPropertyValue("Department", ref department, value);
            }
        }
        [Association]
        public XPCollection<EmployeeTask> Tasks {
            get {
                return GetCollection<EmployeeTask>("Tasks");
            }
        }
    }
}
